### PR TITLE
[WEBGUILD-8624] add motion-unitless.d.ts file

### PR DESCRIPTION
**Pull request by brockthebear**
## Description:

- Add ts-exports-motion-unitless template
- Update seeds-motion/config.json to include `seeds-motion-unitless.d.ts` export

## GitHub pages build link:

- https://[YOUR GITHUB USER NAME HERE].github.io/seeds/

## Jira:

- [WEBGUILD-8624](https://sprout.atlassian.net/browse/WEBGUILD-8624)

## Screenshots:

[WEBGUILD-8624]: https://sprout.atlassian.net/browse/WEBGUILD-8624?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/seeds-motion/config.json
Comment (brockthebear) [inline packages/seeds-motion/config.json:1]: Not sure if the formatting on this file is due to global Prettier config or something I did inadvertently locally.

File: packages/seeds-motion/config.json [hunk 1]
```diff
@@ -1,64 +1,78 @@
...
+					"destination": "seeds-motion.d.ts",
+					"format": "typescript/es6-declarations",
+					"options": {"showFileHeader": false}
+				},
+				{
+					"destination": "seeds-motion-unitless.js",
+					"format": "template/js-exports",
+					"options": {"showFileHeader": false}
+				},
+				{
```
Comment (brockthebear) [inline packages/seeds-motion/config.json:47]: This is the most "significant" change -- adding `seeds-motion-unitless.d.ts`

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     }
   },
   "lint-staged": {
-    "**/*.js": [
+    "**/*.{js,json,ts}": [
```
Comment (iamtheozzy) [inline package.json:32]: THANK YOU!

File: packages/seeds-motion/config.json [hunk 3]
```diff
@@ -34,6 +38,11 @@
           "destination": "seeds-motion.d.ts",
           "format": "typescript/es6-declarations",
           "options": { "showFileHeader": false }
+        },
+        {
+          "destination": "seeds-motion-unitless.d.ts",
```
Comment (iamtheozzy) [inline packages/seeds-motion/config.json:43]: this is a nit pick, but can you nest this object under the `web/js-unitless` property? That way the related files are still grouped together.

Comment (brockthebear) [reply packages/seeds-motion/config.json:43]: Absolutely. I even meant to ask about that. Thanks for catching 👍

File: packages/seeds-motion/config.json [hunk 4]
```diff
@@ -34,6 +38,11 @@
...
+          "format": "template/ts-exports-motion-unitless",
```
Comment (iamtheozzy) [inline packages/seeds-motion/config.json:44]: Curious why you want to create your own custom template vs using the native `typescript/es6-declarations` to generate the declaration file? 

I think the difference is how the types file would then look like this 

```
export const MOTION_EASE_IN : string;
export const MOTION_EASE_OUT : string;
export const MOTION_EASE_INOUT : string;
export const MOTION_DURATION_FAST : number;
export const MOTION_DURATION_MEDIUM : number;
export const MOTION_DURATION_SLOW : number;
```
The main difference would be lack of a default which I think is closer to build output of this package.

**Comment by iamtheozzy** [comment]
I left a couple of comments, but overall all this looks good! 

Something I'm realizing is missing from our readme is the omission on how to generate a changeset, which we need in order to update the package version and changelog. 

Can you run `yarn changeset` and follow the prompts to generate a changeset and commit it?

File: packages/seeds-motion/config.json (hunk 4 above)
Comment (brockthebear) [reply packages/seeds-motion/config.json:44]: Yeah for sure, good question! I tried that first but when I did, I received the output of 
```ts
export const MOTION_EASE_IN : string;
export const MOTION_EASE_OUT : string;
export const MOTION_EASE_INOUT : string;
export const MOTION_DURATION_FAST : string;
export const MOTION_DURATION_MEDIUM : string;
export const MOTION_DURATION_SLOW : string;
```

note that the `MOTION_DURATION_*` values are listed as strings. that's incorrect, as they are actually `number`. I didn't know another way around that issue so I went with the custom template, but if there's an alternate way to get the correct output I'm all for it. Thanks for asking about this! I should have called that out earlier.

**Review by iamtheozzy — APPROVED** [review]

File: packages/seeds-motion/config.json (hunk 4 above)
Comment (iamtheozzy) [reply packages/seeds-motion/config.json]: I just tried this with the `typescript/es6-declarations`

and got the correct output
```tsx
export const MOTION_EASE_IN : string;
export const MOTION_EASE_OUT : string;
export const MOTION_EASE_INOUT : string;
export const MOTION_DURATION_FAST : number;
export const MOTION_DURATION_MEDIUM : number;
export const MOTION_DURATION_SLOW : number;
```

can we try it with this first before we resort to something custom?

Comment (brockthebear) [reply packages/seeds-motion/config.json]: just tried it and i got the correct output as well. moving that config object to be nested under `web/js-unitless` did the trick. thanks a ton for the assist ozzy!

**Review by mchlltt — APPROVED** [review]
Awesome work, thank you!

**Review by iamtheozzy — APPROVED** [review]

**Comment by iamtheozzy** [comment]
@brockthebear thanks for the contribution! Much appreciated! 

We've approved the PR and its ready to merge whenever you want. 

Keep in mind that merging this in will also open a second PR called `Version Packages` that you'll need to approve and merge in as well in order for the updates to be published successfully. 

See https://github.com/sproutsocial/seeds-packets#contributing for more details =]

**Comment by brockthebear** [comment]
> @brockthebear thanks for the contribution! Much appreciated!
> 
> We've approved the PR and its ready to merge whenever you want.
> 
> Keep in mind that merging this in will also open a second PR called `Version Packages` that you'll need to approve and merge in as well in order for the updates to be published successfully.
> 
> See [sproutsocial/seeds-packets#contributing](https://github.com/sproutsocial/seeds-packets#contributing) for more details =]

Hey @iamtheozzy! Thanks a ton for the follow-up! I'll be on the lookout for the second PR. I don't appear to be authorized to merge this PR, however. Is there something I need to do? If not, I'm okay with anybody else merging this PR. 
09-08-2023 at 04 20 15 PM@2x

**Pull request merged**